app/models/internal_post.rb

```diff
@@ -0,0 +1,33 @@
+class InternalPost < Post
+  belongs_to :user
+  validates :user, presence: true
+  has_many :comments, as: :parent
+
+  before_save do
+    parser = Redcarpet::Markdown.new(Redcarpet::Render::HTML, autolink: true)
+    self.body_html = parser.render(body_markdown)
```

Comment:
We should add `validates :body_markdown, presence: true` since we require this field.